src/components/form/DateRangeInput/DateRangeInput.tsx

```diff
@@ -1,14 +1,109 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import _ from 'lodash';
+import type { CSSProperties } from 'react';
+import { isEmpty, isEqual, get, map, extend, pick } from 'lodash';
 import moment from 'moment';
+import type { Moment } from 'moment';
 
 import { Button, Grid, Input, Label, List, Segment } from 'semantic-ui-react';
+import type { ButtonProps, InputProps } from 'semantic-ui-react';
 import ApproveButton from 'components/buttons/ApproveButton';
 import CancelButton from 'components/buttons/CancelButton';
 import Popup from 'components/popups/Popup';
+import type { ApproveButtonProps } from 'components/buttons/ApproveButton/ApproveButton';
 import DateInput from '../DateInput';
 import DatePicker from '../DatePicker';
+import type { DatePickerProps } from '../DatePicker/DatePicker';
+
+interface DateRangeValue {
+    range: string;
+    start: string;
+    end: string;
+}
+
+type RangeValue = Omit<DateRangeValue, 'range'>;
+type Range = Record<string, RangeValue>;
```

Comment:
I'm wondering if we can improve the naming here 🤔 

What do you think about that:
```suggestion
interface Range {
    start: string;
    end: string;
}

interface DateRange extends Range {
    range: string;
}

type Ranges = Record<string, Range>;
```
?

Reply:
Sure, seems like a good suggestion :)

Reply:
Done as a part of [this commit](https://github.com/cloudify-cosmo/cloudify-ui-components/pull/168/commits/ce681665096cbf4e5c18148217a3f81a4457e4d6)